libs/blocks/editorial-card/editorial-card.js

```diff
@@ -0,0 +1,121 @@
+import { createTag, loadStyle, getConfig } from '../../utils/utils.js';
+import { decorateBlockBg, decorateBlockText, decorateBlockHrs, decorateTextOverrides, applyHoverPlay } from '../../utils/decorate.js';
+
+function decorateLockupFromContent(el) {
+  const rows = el.querySelectorAll(':scope > div > p');
+  const firstRowImg = rows[0]?.querySelector('img');
+  if (firstRowImg) {
+    rows[0].classList.add('lockup-area');
+    rows[0].childNodes.forEach((node) => {
+      if (node.nodeType === 3 && node.nodeValue !== ' ') {
+        const newSpan = createTag('span', { class: 'lockup-label' }, node.nodeValue);
+        node.parentElement.replaceChild(newSpan, node);
+      }
+    });
+  }
+}
```

Comment:
The IFs can be flattened and you can use destructuring here
```suggestion
function decorateLockupFromContent(el) {
  const [firstRow] = el.querySelectorAll(':scope > div > p');
  const firstRowImg = firstRow?.querySelector('img');
  if(!firstRowImg) return
  firstRow.classList.add('lockup-area');
  firstRow.childNodes.forEach((node) => {
    if(nodeType !== 3 || node.nodeValue !== '') return
    const newSpan = createTag('span', { class: 'lockup-label' }, node.nodeValue);
    node.parentElement.replaceChild(newSpan, node);
  });
}
```